tsfile/src/main/java/org/apache/iotdb/tsfile/write/writer/TsFileIOWriter.java

```diff
@@ -202,13 +204,19 @@ public void writeChunk(Chunk chunk, ChunkMetaData chunkMetadata) throws IOExcept
    * end chunk and write some log.
    */
   public void endCurrentChunk() {
-    currentChunkGroupMetaData.addTimeSeriesChunkMetaData(currentChunkMetaData);
+    chunkMetaDataList.add(currentChunkMetaData);
+    Path path = new Path(deviceId, currentChunkMetaData.getMeasurementUid());
+    List<ChunkMetaData> chunkMetaDataListOfOnePath = timeseriesMetadataMap.getOrDefault(path,
+        new ArrayList<ChunkMetaData>());
+    chunkMetaDataListOfOnePath.add(currentChunkMetaData);
+    timeseriesMetadataMap.put(path, chunkMetaDataListOfOnePath);
```

Comment:
Please use the following codes instead to reduce the operation and improve the efficiency.
```
timeseriesMetadataMap.computeIfAbsent(path, k -> new ArrayList<>()).add(currentChunkMetaData);
```

Reply:
Fixed.